backends/ebpf/p4c-ebpf.cpp

```diff
@@ -39,6 +39,19 @@ void compile(EbpfOptions& options) {
         ::error("This compiler only handles P4-16");
         return;
     }
+
+    if (options.target != nullptr) {
+        cstring device, arch, vendor;
+        std::tie(device, arch, vendor) = options.parseTarget();
+
+        options.preprocessor_options += " -D__TARGET_EBPF__";
+        if (arch == "ss") {
```

Comment:
This is wrong, you can never compile ebpf programs for v1model or PSA targets.

Reply:
Is it impossible to support psa on the ebpf target/backend?  (psa and v1model are architectures, not targets -- ideally you could compile any architecture on any target, or at least get a clear message that that arhitecture/target combination is not supported).

Reply:
Take a look at p4include/ebpf_model.p4 and it will be obvious that it is very different from v1model.p4. For example, the ebpf back-end only supports packet filters. It does not have output ports or output packets, just a boolean. The set of externs supported is very different. Table properties are different, for example, there are no ternary key matches.